src/bin/vip-app-deploy.ts

```diff
@@ -118,6 +120,11 @@ export async function appDeployCmd( arg: string[] = [], opts: Record< string, un
 	const envId = env.id as number;
 	const track = trackEventWithEnv.bind( null, appId, envId );
 
+	if ( CUSTOM_DEPLOY_KEY ) {
+		debug( 'Validating custom deploy key...' );
+		await validateCustomDeployKey( CUSTOM_DEPLOY_KEY, envId );
```

Comment:
should this just be part of `gates`?

Reply:
Ugh, I think I need to do some restructuring. Maybe I'll move the logic and call it `validateFile()` instead of `gates()`. To answer your question though, I didn't want to add an additional parameter to `gates()`.

Reply:
I don't think it needs another parameter since it is read from `process.env`